be/src/exprs/array_functions.tpp

```diff
@@ -1060,4 +1060,300 @@ private:
+
+class ArrayArithmetic {
+public:
+    template <ArithmeticType type, LogicalType value_type, LogicalType sum_result_type, LogicalType avg_result_type,
+              bool has_null, typename ElementColumn>
+    static StatusOr<ColumnPtr> _sum_and_avg(const ElementColumn& elements, const UInt32Column& offsets,
+                                            const NullColumn::Container* null_elements,
+                                            std::vector<uint8_t>* null_ptr) {
+        const size_t num_array = offsets.size() - 1;
+        auto offsets_ptr = offsets.get_data().data();
+
+        using ValueType = RunTimeCppType<value_type>;
+        using ResultColumnType = std::conditional_t<type == ArithmeticType::SUM, RunTimeColumnType<sum_result_type>,
+                                                    RunTimeColumnType<avg_result_type>>;
+
+        using ResultType = std::conditional_t<
+                type == ArithmeticType::SUM, RunTimeCppType<sum_result_type>,
+                std::conditional_t<std::is_same_v<DateColumn, ResultColumnType>, double,
+                                   std::conditional_t<std::is_same_v<TimestampColumn, ResultColumnType>, double,
+                                                      RunTimeCppType<avg_result_type>>>>;
+
+        auto result_column = ResultColumnType::create();
+        result_column->reserve(num_array);
+
+        auto elements_ptr = (const ValueType*)(elements.raw_data());
+
+        for (size_t i = 0; i < num_array; i++) {
+            size_t offset = offsets_ptr[i];
+            size_t array_size = offsets_ptr[i + 1] - offsets_ptr[i];
+            ResultType sum{};
+
+            bool has_data = false;
+            for (size_t j = 0; j < array_size; j++) {
+                if constexpr (has_null) {
+                    if ((*null_elements)[offset + j] != 0) {
+                        continue;
+                    }
+                }
+
+                has_data = true;
+                auto& value = elements_ptr[offset + j];
+                if constexpr (lt_is_datetime<value_type>) {
+                    sum += value.to_unix_second();
+                } else if constexpr (lt_is_date<value_type>) {
+                    sum += value.julian();
+                } else {
+                    sum += value;
+                }
+            }
+
+            if (has_data) {
+                if constexpr (lt_is_decimalv2<value_type>) {
+                    if constexpr (type == ArithmeticType::SUM) {
+                        result_column->append(sum);
+                    } else {
+                        result_column->append(sum / DecimalV2Value(array_size, 0));
+                    }
+                } else if constexpr (lt_is_arithmetic<value_type> || lt_is_decimal<value_type>) {
+                    if constexpr (type == ArithmeticType::SUM) {
+                        result_column->append(sum);
+                    } else {
+                        result_column->append(sum / array_size);
```

Comment:
for decimal v3, here seems incorrect, for an example:

expr array_avg(cast(1 as decimal(18,0)), cast(1 as decimal(18,0)), cast(2 as decimal(18,0))) would yield 
1, but a meaningful result should be 1.333333

Reply:
yeah, your are right, I will refact array_sum/array_avg for decimalV3 late, there just move template